javascript/ql/lib/semmle/javascript/StandardLibrary.qll

```diff
@@ -115,7 +115,17 @@ class StringReplaceCall extends DataFlow::MethodCallNode {
    * Holds if this is a global replacement, that is, the first argument is a regular expression
    * with the `g` flag, or this is a call to `.replaceAll()`.
    */
-  predicate isGlobal() { this.getRegExp().isGlobal() or this.getMethodName() = "replaceAll" }
+  predicate isGlobal() {
+    RegExp::isGlobal(this.getRegExp().getFlags()) or this.getMethodName() = "replaceAll"
+  }
+
+  /**
+   * Holds if this is a global replacement, that is, the first argument is a regular expression
+   * with the `g` flag, or this is a call to `.replaceAll()`.
+   */
```

Comment:
This docstring needs to mention that the predicate also holds if the flags are unknown.  
Incorporate that into the string naturally in some way.  
(Actually, try to copy-paste the above sentence into Copilot as instructions, use `o1-mini` or `o1-preview` with the Copilot edit mode: Select the docstring in VSCode, press `cmd + i`, make sure to select the right model, and paste my above instructions).

Reply:
Fixed 8b64f17c804da2d0866d419a7d2a12679568fce1